ompi/datatype/ompi_datatype.h

```diff
@@ -374,5 +375,25 @@ OMPI_DECLSPEC int ompi_datatype_unpack_external( const char datarep[], const voi
 OMPI_DECLSPEC int ompi_datatype_pack_external_size( const char datarep[], int incount,
                                                     ompi_datatype_t *datatype, MPI_Aint *size);
 
+#define OMPI_DATATYPE_RETAIN(ddt)                                       \
+    {                                                                   \
+        if( !ompi_datatype_is_predefined((ddt)) ) {                     \
+            OBJ_RETAIN((ddt));                                          \
+            OPAL_OUTPUT_VERBOSE((0, 100, "Datatype %p [%s] refcount %d in file %s:%d\n",     \
+                                (void*)(ddt), (ddt)->name, (ddt)->super.super.obj_reference_count, \
+                                __FILE__, __LINE__));                   \
+        }                                                               \
+    }
+
+#define OMPI_DATATYPE_RELEASE(ddt)                                      \
+    {                                                                   \
+        if( !ompi_datatype_is_predefined((ddt)) ) {                     \
+            OPAL_OUTPUT_VERBOSE((0, 100, "Datatype %p [%s] refcount %d in file %s:%d\n",     \
```

Comment:
This message reports non-thread safe data, with the exception of developers nobody should ever use it. Let's keep it as is.